modules/dnn/src/layers/cpu_kernels/fast_norm.cpp

```diff
@@ -0,0 +1,133 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "../../precomp.hpp"
+#include "fast_norm.hpp"
+
+namespace cv { namespace dnn {
+
+//  Mean-Variance-Normalization speedup by multi-threading.
+void fastNorm(const Mat &input, Mat &output, float epsilon, int normalized_axis, bool normalize_variance) {
+    const auto input_shape = shape(input);
+    CV_CheckGE(normalized_axis, 0, "fastNorm: axis out of range");
+    CV_CheckLT(normalized_axis, static_cast<int>(input_shape.size()), "fastNorm: axis out of range");
+
+    size_t loops = static_cast<size_t>(total(input_shape, 0, normalized_axis)),
+           norm_size = static_cast<size_t>(total(input_shape, normalized_axis));
+    float inv_norm_size = 1.0 / norm_size;
+
+    auto fn = [&](const Range &r) {
+        const auto *input_data = input.ptr<const float>();
+        auto *output_data = output.ptr<float>();
+        for (int i = r.start; i < r.end; i++) {
+            const auto *x = input_data + norm_size * i;
+            auto *y = output_data + norm_size * i;
+
+            float mean = 0.f, mean_square = 0.f;
+            for (int j = 0; j < norm_size; j++) {
+                float v = x[j];
+                mean += v;
+                mean_square += v * v;
+            }
+
+            mean *= inv_norm_size;
+            mean_square = std::sqrt(std::max(0.f, mean_square * inv_norm_size - mean * mean) + epsilon);
+            float inv_stdev = normalize_variance ? 1.f / mean_square : 1.f;
```

Comment:
Seems like we don't need to compute `mean_square` if `normalize_variance=false`

Reply:
It is true but given the fact that `normalize_variance=false` is rare case so I guess this implementation is acceptable.